markdown/m2h/handlers/index.ts

```diff
@@ -27,35 +27,61 @@ function getL10nCardMap(locale = DEFAULT_LOCALE) {
 
   Object.keys(listMsgObj).forEach((msgName) => {
     l10nCardMap.set(
-      listMsgObj[msgName]["msgstr"][0],
-      listMsgObj[msgName]["msgid"]
+      listMsgObj[msgName]["msgid"],
+      listMsgObj[msgName]["msgstr"][0]
```

Comment:
This allows for a much easier lookup using GFM notecard keywords.  The logic is reversed when using MDN-flavored syntax, so it iterates through each entry to find the applicable type.

Reply:
Would it make sense to add (key, value) and (value, key) to the map to allow both directions, or is there a risk for a collision? In either case, let's add a comment above to give an example what msgid and msgstr[0] refer to to help.

Also, let's update the JSDoc of the method to clearly indicate what the returning map contains.

Reply:
It might make sense to have variables for both formats, but I feel that if we are planning to [migrate to GFM syntax](https://github.com/mdn/yari/pull/10168#issuecomment-2037083067), then I think it makes more sense to just have the one format, as it will make removal of the old syntax parsing much quicker.